File: TowerOFHanoi.c
#include<stdio.h>

void towerOfHanoi(int n,char src,char helper,char des){
    if(n==1){
        printf("\nTransfer disk %d from %c to %c.",n,src,des);
        return;
    }
    towerOfHanoi(n-1,src,des,helper);
    printf("\nTransfer disk %d from %c to %c.",n,src,des);
    towerOfHanoi(n-1,helper,src,des);
}

int main()
{
    int n=3;
    towerOfHanoi(n,'A','B','C');
    return 0;
}

File: TowerOfHanoi.java
public class TowerOfHanoi{
    public static void towerOfHanoi(int n, String source,String helper ,String destination){

        if(n==1){
        System.out.println("Transfer disk "+n+" from "+ source +" to "+destination);
        return;
        }
        towerOfHanoi(n-1, source, destination, helper);
        System.out.println("Transfer disk "+n+" from "+ source +" to "+destination);
        towerOfHanoi(n-1, helper, source, destination);
    }
    public static void main(String args[]){
        int n=10;
        towerOfHanoi(n,"S","H","D");
    }
}